src/host.ps1

```
@@ -0,0 +1,162 @@
+                
+                if ($hostUrls -and $hostUrls.Count -gt 0) {
+                    Write-Verbose "Found $($hostUrls.Count) hosts"
+                    
+                    # Apply limit if specified
+                    $processLimit = if ($Limit -gt 0) { [Math]::Min($Limit, $hostUrls.Count) } else { $hostUrls.Count }
+                    $processUrls = $hostUrls | Select-Object -First $processLimit
+                    
+                    foreach ($hostUrl in $processUrls) {
+                        # Skip null or empty URLs
+                        if ([string]::IsNullOrEmpty($hostUrl)) {
+                            continue
+                        }
+                        
+                        try {
+                            # Extract the ID from the URL string
+                            $match = [regex]::Match($hostUrl, "\/rest\/v0\/hosts\/([^\/]+)$")
+                            if ($match.Success) {
+                                $id = $match.Groups[1].Value
+                                if (![string]::IsNullOrEmpty($id)) {
+                                    $hostDetail = Invoke-RestMethod -Uri "$script:XoHost$hostUrl" @script:XoRestParameters
+
+                                    if ($hostDetail) {
+                                        # Parse the response if it's a string
+                                        if ($hostDetail -is [string]) {
+                                            $hostDetail = $hostDetail | ConvertFrom-Json -AsHashtable
+                                        }
+                                        
+                                        ConvertTo-XoHostObject $hostDetail
+                                    }
+                                }
+                                else {
+                                    Write-Warning "Failed to extract valid ID from URL: $hostUrl"
```

should throw instead of Write-Warning, which will also flatten the logic by quite a bit; same for all the other fetch warnings